packages/aws-rfdk/lib/lambdas/nodejs/x509-certificate/types.ts

```diff
@@ -165,6 +171,10 @@ export function implementsIX509CertificateGenerate(value: any): boolean {
   if (!implementsIX509ResourceProperties(value)) { return false; }
   if (!implementsDistinguishedNameProps(value.DistinguishedName)) { return false; }
   if (value.SigningCertificate && !implementsISecretCertificate(value.SigningCertificate)) { return false; }
+  if (value.CertificateValidFor) {
+    if (typeof(value.CertificateValidFor) !== 'string') { return false; }
+    if (value.CertificateValidFor === '' || isNaN(Number(value.CertificateValidFor))) { return false; }
```

Comment:
Not here, but it makes some sense to assert that in the construct itself. Fail early...

Reply:
I'm okay with this approach, although it does miss the case where the customer passes in a CDK token. Sounds like that should be avoided since, as this PR points out, it could lead to certificates being regenerated and breaking any manually configured clients.

On that basis, do you think we should also validate that the expiration is not a token?

Reply:
Token should be allowable. That'll get expanded into its value by CloudFormation.

Though, I suppose that means that my validation check isn't quite right. :-/

Reply:
Updated in the latest revision. I'm leaving the negative-value test out of the type validator. Those validators do not give useful diagnostic output; they're mostly to make sure that we've created the CustomResource properly.

openssl will die with an error like `req: Non-positive number "-1" for -days` if a negative value is passed through.

Reply:
Great - thanks for looking into this.